### PR TITLE
Added functions for crops and ag model API endpoints

### PR DESCRIPTION
**Pull request by colinloftin-awhere**

### CODE REVIEW

File: R/models.R [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+    return(as.data.frame(data))
+  }
+}
+
+
+
+
+
+
+#' @title Get Model Results
```
Comment (drewmarticorenaAWHERE) [inline R/models.R:182]: This doesn't work for me.  I get back a different named list in "a" then you are expecting when I use this code below
[credentials.txt](https://github.com/aWhereAPI/aWhere-R-Library/files/3560341/credentials.txt)


library(aWhereAPI)

load_credentials('c:/aWhere/credentials/credentials.txt')

get_fields()
get_planting('field_test4')

get_model_results(field_id = 'field_test4'
                  ,model_id = 'Corn2500ISUAbendroth')

File: R/models.R [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    
+    a <- suppressMessages(httr::content(request))
+    
+    doWeatherGet <- aWhereAPI:::check_JSON(a,request)[[1]]
+  }
+  
+  
+  a[sapply(a, is.null)] <- NA
+  
+  data <- data.frame(biofixDate = a$biofixDate, nextStage = as.character(a$nextStage), gddUnits = a$gddUnits, modelId = a$modelId, latitude = a$location$latitude,
```
Comment (drewmarticorenaAWHERE) [inline R/models.R]: when I do the above commands this is the contents of the named list

names(a)
[1] "biofixDate"     "previousStages" "currentStage"   "nextStage"      "gddUnits"       "modelId"        "location"       "plantingDate"  
[9] "_links"

**Review by drewmarticorenaAWHERE — COMMENTED** [review]
Everything works except the model results

**Comment by colinloftin-awhere** [comment]
@drewmarticorenaAWHERE I believe I've fixed the issue you found. It wasn't appropriately handling the different stages that returned in the response. I tested it on all crop/model combinations and it seems to be working correctly now.

**Pull request merged**